main.tf

```diff
@@ -48,6 +48,8 @@ locals {
   # attach_ibm_managed_security_group is false and custom_security_group_ids is not set => default behavior, so set to null
   # attach_ibm_managed_security_group is false and custom_security_group_ids is set => only use the custom security group ids
   cluster_security_groups = var.attach_ibm_managed_security_group == true ? (var.custom_security_group_ids == null ? null : concat(["cluster"], var.custom_security_group_ids)) : (var.custom_security_group_ids == null ? null : var.custom_security_group_ids)
+  # tflint-ignore: terraform_unused_declarations
+  validate_private_endpoint = var.use_private_endpoint == true && var.cluster_config_endpoint_type == "default" ? tobool("When setting 'var.use_private_endpoint' to true, 'var.cluster_config_endpoint_type' must be set to private.") : true
```

Comment:
We can double check with @toddgiguere but I'm not sure "default" = public endpoint. For example, if you have a private only cluster, I think "default" would be private (but I might be wrong)